### PR TITLE
rc: use new client from sdk

### PR DESCRIPTION
**Pull request by edaniels**

### CODE REVIEW

File: web/frontend/src/components/gamepad.vue [hunk 1]
```diff
@@ -165,36 +176,45 @@ const tick = () => {
...
-  curStates.LT = trunc(gamepad.buttons[4]!.value);
-  curStates.RT = trunc(gamepad.buttons[5]!.value);
-  curStates.Select = trunc(gamepad.buttons[8]!.value);
-  curStates.Start = trunc(gamepad.buttons[9]!.value);
-  curStates.LThumb = trunc(gamepad.buttons[10]!.value);
-  curStates.RThumb = trunc(gamepad.buttons[11]!.value);
-  curStates.Menu = trunc(gamepad.buttons[16]!.value);
+  // TODO(RSDK-881): this ought to detect actual controller mappings; for now
+  // just try to not fail.
+  curStates.X = trunc(gamepad.axes[0]);
```
Comment (edaniels) [inline web/frontend/src/components/gamepad.vue:199]: fixes!

File: web/frontend/src/app.vue [hunk 2]
```diff
@@ -1,5 +1,6 @@
 <!-- eslint-disable require-atomic-updates -->
 <script setup lang="ts">
+
```
Comment (edaniels) [inline web/frontend/src/app.vue:3]: this actually contains a lot of driveby bug fixes that improves RC reliability

File: web/cmd/directremotecontrol/main.go [hunk 3]
```diff
@@ -0,0 +1,151 @@
+/*
+Package main provides a utility to run a remote control only web server. It is configured with
```
Comment (edaniels) [inline web/cmd/directremotecontrol/main.go:2]: this helps test RC against app

**Review by maximpertsov — APPROVED** [review]

File: web/runtime-shared/templates/webappindex.html [hunk 4]
```diff
@@ -25,23 +25,26 @@
     {{ else if eq .Env "development" }}
     <script type="module">window.PRIME_CONFIG = { base: '/static' }</script>
     {{ end }}
+
+    <script id="_timeout_worker" type="javascript/worker">
```
Comment (edaniels) [inline web/runtime-shared/templates/webappindex.html:29]: saving this for something special :)

File: web/frontend/src/components/base.vue [hunk 5]
```diff
@@ -280,7 +284,7 @@ onMounted(() => {
             class="mb-2"
             @keydown="handleKeyDown"
             @keyup="handleKeyUp"
-            @toggle="(active: boolean) => !active && stop()"
+            @toggle="(active: boolean) => { !active && (pressed.size > 0 || !stopped) && stop() }"
```
Comment (edaniels) [inline web/frontend/src/components/base.vue:287]: more fixes for off focus

File: web/frontend/src/app.vue [hunk 6]
```diff
@@ -64,7 +65,7 @@ const relevantSubtypesForStatus = [
   'input_controller',
 ];
 
-const passwordInput = $ref<HTMLInputElement>();
+const password = $ref<string>('');
```
Comment (edaniels) [inline web/frontend/src/app.vue:68]: this helps password auth work again

File: web/frontend/src/app.vue [hunk 7]
```diff
@@ -481,10 +475,11 @@ const createConnectionManager = () => {
 
           statuses.ops = true;
         } catch (error) {
-          if (error instanceof ConnectionClosedError) {
+          if (ConnectionClosedError.isError(error)) {
```
Comment (edaniels) [inline web/frontend/src/app.vue:478]: better errors

File: web/frontend/src/app.vue [hunk 8]
```diff
@@ -511,12 +508,16 @@ const createConnectionManager = () => {
...
         resourcesOnce = false;
 
-        await window.connect();
+        await client.connect();
         await fetchCurrentOps();
         lastStatusTS = Date.now();
         console.log('reconnected');
       } catch (error) {
-        console.error('failed to reconnect; retrying:', error);
+        if (ConnectionClosedError.isError(error)) {
```
Comment (edaniels) [inline web/frontend/src/app.vue:516]: less spam

File: web/frontend/src/app.vue [hunk 9]
```diff
@@ -542,25 +543,25 @@ const createConnectionManager = () => {
...
   return status[resourceNameToString(resource)];
 };
 
-const rawResourceStatusByName = (resource: Resource) => {
+const rawResourceStatusByName = (resource: commonApi.ResourceName.AsObject) => {
   return rawStatus[resourceNameToString(resource)];
 };
 
-const hasWebGamepad = () => {
+const filteredWebGamepads = () => {
```
Comment (edaniels) [inline web/frontend/src/app.vue:556]: more gamepads

File: web/frontend/src/app.vue [hunk 10]
```diff
@@ -743,23 +742,26 @@ onMounted(async () => {
...
-          @keyup.enter="doLogin(authType)"
-        >
-        <v-button
-          :disabled="disableAuthElements"
-          label="Login"
-          @click="disableAuthElements ? undefined : doLogin(authType)"
-        />
+      <div class="px-4 py-3">
+        <span>{{ authType }}: </span>
+        <div class="w-96">
```
Comment (edaniels) [inline web/frontend/src/app.vue:747]: fixes auth again
